Sources/TuistGenerator/Models/Scheme.swift

```diff
@@ -170,3 +174,39 @@ public class RunAction: Equatable {
             lhs.arguments == rhs.arguments
     }
 }
+
+public class ArchiveAction: Equatable {
+    // MARK: - Attributes
+
+    public let configurationName: String
+    public let revealArchiveInOrganizer: Bool
+    public let customArchiveName: String?
+    public let preActions: [ExecutionAction]
+    public let postActions: [ExecutionAction]
+
+    // MARK: - Init
+
+    public init(
+        configurationName: String,
+        revealArchiveInOrganizer: Bool = true,
+        customArchiveName: String? = nil,
+        preActions: [ExecutionAction] = [],
+        postActions: [ExecutionAction] = []
+    ) {
+        self.configurationName = configurationName
+        self.revealArchiveInOrganizer = revealArchiveInOrganizer
+        self.customArchiveName = customArchiveName
+        self.preActions = preActions
+        self.postActions = postActions
+    }
+
+    // MARK: - Equatable
+
+    public static func == (lhs: ArchiveAction, rhs: ArchiveAction) -> Bool {
```

Comment:
will this not synthesise automatically?

Reply:
The compiler only synthesizes `Equatable` for value types, like struct. I don't se a problem in using a struct instead of a class. I'd be glad to change the `Actions` to structs if you wish.

Reply:
Ah, I missed it was a class - thats fine.